src/huggingface_hub/keras_mixin.py

```diff
@@ -127,15 +134,15 @@ def _create_model_card(
 
     model_card += "\n ## Training Metrics"
     model_card = _write_metrics(model, model_card)
-    model_card += "\n ## Model Plot\n"
-    model_card += "\n<details>"
-    if model_plot:
+    if model_plot and os.path.exists(f"{repo_dir}/model.png"):
+        model_card += "\n ## Model Plot\n"
+        model_card += "\n<details>"
         model_card += "\n<summary>View Model Plot</summary>\n"
         path_to_plot = "./model.png"
         model_card += f"\n![Model Image]({path_to_plot})\n"
     model_card += "\n</details>"
 
-    if os.path.exists(readme_path):
+    if os.path.exists(readme_path) and from_callback is False:
```

Comment:
Am I missing something here or are we reading a file at `readme_path` and then writing the same contents right back to `readme_path` again? If that's the case, you can just ignore in this case and just write if the content is different.

If there was a different path you're reading from in order to write, that would make sense (and was sorta what I first expected when I saw this)

Reply:
I read it again and thought it was quite funny, I think it was there initially and I didn't want to change the behavior much not to break it that's why I modified it like that lol